src/mscorlib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -777,6 +775,66 @@ public int EnsureCapacity(int capacity)
             return newSize;
         }
 
+        /// <summary>
+        /// Sets the capacity of this dictionary to hold up at least all existing entries
+        /// 
+        /// This method can be used to minimize the memory overhead 
+        /// once it is known that no new elements will be added. 
+        /// 
+        /// To completely clear a dictionary and release all memory 
+        /// referenced by the dictionary, execute the following statements:
+        /// 
+        /// dictionary.Clear();
+        /// dictionary.TrimExcess();
+        /// </summary>
+        public void TrimExcess()
+        {
+            TrimExcess(HashHelpers.GetPrime(Count));
```

Comment:
Did we decide whether to GetPrime or use Count directly? I guess GetPrime will give better performance, so as you've written it, this overload doesn't lead to the "pit of failure"

cc @benaadams

Reply:
It has to be GetPrime for correctness. If it is not a prime, you can end up with infinite loops.

Reply:
Ah yes it's just missing below.